lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -149,6 +174,80 @@ def remove_schedule(name)
 
       private
 
+      def supports_lua?
+        redis_master_version >= 2.5
+      end
+
+      def redis_master_version
+        Resque.redis.info['redis_version'].to_f
+      end
+
+      def startup_sha(refresh = false)
+        @startup_sha = nil if refresh
+
+        @startup_sha ||=
+          Resque.redis.script(:load, <<-EOF.gsub(/^ {14}/, ''))
+            -- KEYS
+            -- 1: schedules
+            -- 2: persisted_schedules
+            -- 3: schedules_changed
+            -------------------------
+            -- ARGV
+            -- i:   schedule_name
+            -- i+1: encoded_schedule_config
+            -- i+2: is_schedule_persisted?
+
+            local remove_schedule = function(schedule)
+              redis.call("HDEL", KEYS[1], schedule)
+              redis.call("SREM", KEYS[2], schedule)
+              redis.call("SADD", KEYS[3], schedule)
+            end
+
+            local get_non_persistent_schedules = function()
+              local all_schedules = redis.call("HKEYS", KEYS[1])
+              local non_persistent_schedules = {}
+              for _,schedule in ipairs(all_schedules) do
+                local schedule_is_persisted = redis.call("SISMEMBER", KEYS[2], schedule)
+                if schedule_is_persisted == 0 then
+                  table.insert(non_persistent_schedules, schedule)
+                end
+              end
+              return non_persistent_schedules
+            end
+
+            local set_schedule = function(name, config, persisted)
+              redis.call("HSET", KEYS[1], name, config)
+              redis.call("SADD", KEYS[3], name)
+              if persisted == "true" then
+                redis.call("HSET", KEYS[1], name, config)
```

Comment:
Shouldn't this be `redis.call("SADD", KEYS[2], name)`? So `persisted_schedules` instead of `schedules`, since you already added it to/updated the schedules hash on line 219 with exactly the same attributes.